src/live-programming.rst

```diff
@@ -47,3 +47,210 @@ TODO: in music
+
+Other Applications
+------------------
+
+Programmable physical environments have appeared in many fields. It is hard to say when such concept is first generated. In 1970s, researchers have proposed the idea of programmable industrial automation, which enable factories to efficiently complete work. Though some programmable physical systems don't provide a very flexible interface for people, they are very close to the definition of programmable physical environment.  Here will list some application of programmable physical environment or systems in different fields.
+
+Educational Use
+~~~~~~~~~~~~~~~
+
+One common application of programmable physical system is in the education field, especially for children. LEGO programmable brick is a tool that enable children to apply simple programs to LEGO bricks. The whole complicated programs are wrapped in a device and by working with instructions showed by the device, children, who usually don't have any knowledge of programming, can set some behaviours for LEGO bricks or motors.
+Some research dig more about the programmable bricks and make more developments.
+
+.. container:: bib-item
+
+  .. bibliography:: programmable_PE2.bib
+    :filter: key == "gindling1995lego"
+
+Factory Manufacture
+~~~~~~~~~~~~~~~~~~~
+
+The automation of industrial manufacture has a long history of development. In the 20th century, industrial automation was well advanced in two areas of manufacture, including homogeneous products and mass manufacture of discrete products. In old time, the industrial automation was still a kind of "hard" or "fixed" automation, which doesn't enable much flexibility.
+
+.. container:: bib-item
+
+  .. bibliography:: programmable_PE2.bib
+    :filter: key == "saraswat1994factory"
+
+.. container:: bib-item
+
+  .. bibliography:: programmable_PE3.bib
+    :filter: key == "nitzan1976industrial"
+
+As the development of technology, industrial automation enables more flexible work, which enables the same automated equipment to be programmed to perform different tasks. The development of the controller systems gave the growth of research on designing corresponding interface for the controller systems. Here are some papers on different interface architecture for programmable controller systems.
+
+For now, there are many researches on programmable logic circuit. For college learning and industrial production, software for programming logic circuits is widely used.
+
+However, it is quite hard to divide these application as a branch of programmable physical environment. Most of the applications in factory manufacture are surely including programmable physical things, but there's merely elements of an "environment". The interaction between people and the systems is still limited, especially for the sensing part. The liveness of such systems may be still at a low level, as in industrial automation, it is hard to rapidly see the results from the changes to the programs.
```

Comment:
Good point. I'd suggest wording it as "Although industrial robots are programmable, doing so remains difficult. Making environments shared between humans and robots more programmable remains an area of ongoing and future research."